server/datastore/mysql/software.go

```diff
@@ -2872,243 +2278,753 @@ last_vpp_install AS (
+            LEFT JOIN
+                host_vpp_software_installs hvsi2 ON hvsi.host_id = hvsi2.host_id AND
+                    hvsi.adam_id = hvsi2.adam_id AND
+                    hvsi.platform = hvsi2.platform AND
+                    hvsi2.removed = 0 AND
+                    (hvsi.created_at < hvsi2.created_at OR (hvsi.created_at = hvsi2.created_at AND hvsi.id < hvsi2.id))
+            WHERE
+                hvsi.host_id = :host_id AND
+                hvsi.removed = 0 AND
+                hvsi2.id IS NULL AND
+                NOT EXISTS (
+                    SELECT 1
+                    FROM
+                        upcoming_activities ua
+                    INNER JOIN
+                        vpp_app_upcoming_activities vaua ON ua.id = vaua.upcoming_activity_id
+                    WHERE
+                        ua.host_id = hvsi.host_id AND
+                        vaua.adam_id = hvsi.adam_id AND
+                        vaua.platform = hvsi.platform AND
+                        ua.activity_type = 'vpp_app_install'
+                )
+        )
+        SELECT
+			*
+		FROM
+            (SELECT * FROM upcoming_vpp_install UNION SELECT * FROM last_vpp_install) AS lvia
+    `, vppAppHostStatusNamedQuery("hvsi", "ncr", "status"))
+	vppInstallsStmt, args, err = sqlx.Named(vppInstallsStmt, namedArgs)
+	if err != nil {
+		return nil, nil, ctxerr.Wrap(ctx, err, "build named query for list host software")
+	}
+	var vppInstalls []*hostSoftware
+	err = sqlx.SelectContext(ctx, ds.reader(ctx), &vppInstalls, vppInstallsStmt, args...)
+	if err != nil {
+		return nil, nil, err
+	}
+	byVPPAdamID := make(map[string]*hostSoftware)
+	for _, s := range vppInstalls {
+		if s.VPPAppAdamID != nil {
+			byVPPAdamID[*s.VPPAppAdamID] = s
 		}
+	}
 
-		// promote the VPP app id and version to the proper destination fields
-		if hs.VPPAppAdamID != nil {
-			var version string
-			if hs.VPPAppVersion != nil {
-				version = *hs.VPPAppVersion
-			}
-			hs.AppStoreApp = &fleet.SoftwarePackageOrApp{
-				AppStoreID:  *hs.VPPAppAdamID,
-				Version:     version,
-				SelfService: hs.VPPAppSelfService,
-				IconURL:     hs.VPPAppIconURL,
-			}
+	var stmtAvailable string
 
-			// promote the last install info to the proper destination fields
-			if hs.LastInstallInstallUUID != nil && *hs.LastInstallInstallUUID != "" {
-				hs.AppStoreApp.LastInstall = &fleet.HostSoftwareInstall{
-					CommandUUID: *hs.LastInstallInstallUUID,
-				}
-				if hs.LastInstallInstalledAt != nil {
-					hs.AppStoreApp.LastInstall.InstalledAt = *hs.LastInstallInstalledAt
-				}
+	if (opts.OnlyAvailableForInstall && !opts.VulnerableOnly) || (opts.IncludeAvailableForInstall && !opts.VulnerableOnly) {
+		namedArgs["vpp_apps_platforms"] = fleet.VPPAppsPlatforms
+		if fleet.IsLinux(host.Platform) {
+			namedArgs["host_compatible_platforms"] = fleet.HostLinuxOSs
+		} else {
+			namedArgs["host_compatible_platforms"] = []string{host.FleetPlatform()}
+		}
+		stmtAvailable = `
```

Comment:
Need to fact check this query. Does it need to contain all this stuff to pull up "available to install software". Can we break it down into two pieces software/vpp to reduce complexity.